crates/ruff/src/linter.rs

```diff
@@ -374,7 +374,7 @@ pub fn lint_only(
         &directives,
         settings,
         noqa,
-        source_kind,
+        Some(source_kind),
```

Comment:
Not sure why this is optional, but wanted to avoid increasing the scope of this even more.

Reply:
The reason this is optional is because there's no `--add-noqa` support for Jupyter Notebooks yet. The logic for `add_noqa_to_path` basically calls into `check_path` and as far as I recall there were some circular dependency problems. Maybe it could be solved now with the recent changes.

Reply:
Could `add_noqa_to_path` detect the `PySourceType` and exit early (with a warning) if it is a Jupyter notebook to avoid that all downstream code has to handle optional source kinds?